packages/devtools_app/lib/src/shared/memory/retaining_path.dart

```diff
@@ -57,22 +59,25 @@ class PathFromRoot {
   factory PathFromRoot.forObject(
     HeapSnapshotGraph graph, {
     required List<int> shortestRetainers,
-    required int objectId,
-    required int heapRootIndex,
+    required int index,
   }) {
-    var nextObjectId = shortestRetainers[objectId];
-    if (nextObjectId == heapRootIndex) {
+    HeapClassName objectClass(int index) {
+      final classId = graph.objects[index].classId;
+      return HeapClassName.fromHeapSnapshotClass(graph.classes[classId]);
+    }
+
+    var nextIndex = shortestRetainers[index];
+    if (nextIndex == heapRootIndex) {
       return empty;
     }
 
     final path = <HeapClassName>[];
 
-    while (shortestRetainers[nextObjectId] != heapRootIndex) {
-      nextObjectId = shortestRetainers[nextObjectId];
-      final classId = graph.objects[nextObjectId].classId;
-      final className =
-          HeapClassName.fromHeapSnapshotClass(graph.classes[classId]);
+    while (true) {
```

Comment:
instead of using while true, do `while (nextIndex != heapRootIndex)`. This makes it more clear what the condition is that we are while looping for